src/plugins/intel_cpu/src/nodes/matmul.cpp

```diff
@@ -110,8 +110,75 @@ bool MatMul::isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op,
     return true;
 }
 
+class MMShapeInfer : public ShapeInferEmptyPads {
+public:
+    MMShapeInfer(const std::shared_ptr<const ngraph::opset1::MatMul>& op) : m_matmul(op) {
+        m_out_rank = m_matmul->get_output_partial_shape(0).rank().get_length();
+        m_YShape = VectorDims(m_out_rank, 1); // for output and cache
+    }
+    std::vector<VectorDims> infer(
+        const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
+        const std::unordered_map<size_t, MemoryPtr>& data_dependency) override {
+        VectorDims AShape = input_shapes[0].get();
+        VectorDims BShape = input_shapes[1].get();
```

Comment:
```suggestion
        const VectorDims& shapeA = input_shapes[0].get();
        const VectorDims& shapeB = input_shapes[1].get();
```
When you declare just `VectorDims` copy constructor is called. But we need just to read them, so taking the reference is better.

Reply:
> When you declare just `VectorDims` copy constructor is called. But we need just to read them, so taking the reference is better.

Done